lumicks/pylake/file.py

```diff
@@ -114,10 +123,24 @@ def print_group(group, name="", indent=-2):
                 if isinstance(item, h5py.Dataset):
                     r += print_dataset(item, key, indent + 2)
                 else:
-                    r += print_group(item, key, indent + 2)
+                    if key not in blacklist:
+                        r += print_group(item, key, indent + 2)
             return r
 
-        return print_attributes(self.h5) + "\n" + print_group(self.h5)
+        def print_dicts(field_name):
+            field = getattr(self, field_name)
+            return f'\n.{field_name}\n' + ''.join(f'  - {key}\n' for key in field.keys()) if field else ''
+
+        def print_force(field_name):
+            field = getattr(self, field_name)
+            calibration = '  .calibration\n' if field.calibration else ''
+            return f'.{field_name}\n{calibration}' if field else ''
+
+        return print_attributes(self.h5) + "\n" + print_group(self.h5) + \
+            ''.join((print_dicts(field) for field in ['kymos', 'fdcurves', 'scans', 'markers'])) + "\n" + \
+            ''.join((print_force(field) for field in ['force1x', 'force1y', 'force2x', 'force2y',
+                                                      'force3x', 'force3y', 'force4x', 'force4y']))
```

Comment:
There's some unnecessary duplication here. The `blacklist` already knows about the marker fields. It would be great if you could put all the into a single mapping so that we can just add a field there in order to get the correct printing and `__getitem__` behavior in the future.

Reply:
Good point. I will take the redirected ones from the list directly. I will also swap the dict out for an ordered one to ensure order stability in the print statement (as this is also explicitly tested). I know that in 3.7 dictionaries are ordered by the language standard, but in 3.6 it still seems to be an implementational detail (even though they are ordered for CPython), hence probably best to be explicit about order preservation.

The force fields are a bit special, since we don't really need to discourage people from using those directly. I think for now, having these two separate cases is ok force fields versus the API classes? If we get more complicated redirects, I can expand the logic.